### ISSUE TITLE
bug in rmse calculation; actually computes mae

### ISSUE THREAD

**Issue by jcrdubois** [fooof-tools/fooof#151]
in fooof/fit.py, lines 928-931:

```
def _calc_rmse_error(self):
        """Calculate root mean squared error of the full model fit."""
        self.error_ = np.sqrt((self.power_spectrum - self.fooofed_spectrum_) ** 2).mean()
```  

seems wrong -- you're missing a set of parentheses. What it ends up computing as written is the MAE, mean absolute error.

I would suggest editing the function name to reflect this (e.g. to `_calc_mae_error` if MAE is desired), or editing the formula as per the following:

```
def _calc_rmse_error(self):
        """Calculate root mean squared error of the full model fit."""
        self.error_ = np.sqrt(((self.power_spectrum - self.fooofed_spectrum_) ** 2).mean())
```

Not a big deal, just caught my eye.

**Comment by TomDonoghue** [comment]
Ah, yes, thank you - good catch! You are totally right, we missed a parenthesis here, making it actually calculate MAE. Ooops. It seems that since we have already been using MAE, and it should be an appropriate error metric, we should just fix the descriptions here. 

@rdgao & @voytek  - I'm cc'ing you in here to double check / sign-off that MAE is a reasonably error metric, and since we're (accidentally) already doing it, that we should just keep doing so (we talked about this briefly at some point). 

Also, since the current form is MAE, but in a somewhat roundabout way, a more canonical way to write it would be like this:

```
def _calc_mae_error(self):
    """Calculate mean absolute error of the full model fit."""
    self.error_ = np.abs(self.power_spectrum - self.fooofed_spectrum_).mean()
```

If keeping MAE as our error metric and re-writting docs & implementation as above is cool with all, then we're good to go. I've added this as an update to #152, so it can be merged there if we are all happy with this.

**Comment by voytek** [comment]
Since we're not using these errors to do any form of model selection, in the end it doesn't _really_ matter at this stage. Reporting error is a mostly a convenience at this stage, no?

Intuitively, MAE does not penalize larger "errors", whereas (R)MSE will more strongly penalize them. At that point, the choice becomes a matter of model selection goal. Since we aren't using these for selection (yet), just stick with MAE.

**Comment by TomDonoghue** [comment]
Yeh, we don't use this error anywhere really - it's not used in model fitting or selection. It's just a post-hoc report for people to eyeball if they want to. 

But - in the model fitting, through `curve_fit`, that process does use a least squares error metric. So is it a bit weird if keep MAE in that we use a least squares fit but then report a different error metric?

**Comment by voytek** [comment]
Personally I don't think so, since curve_fit is using it for parameter selection and we _do_ want to penalize larger errors there. And it's not immediately obvious to casual users that this difference exists, or that it matters. And for experts, I don't think it would much matter, either.

Again, personally, I wouldn't even bat an eye at this.

**Comment by rdgao** [comment]
just to clarify: the model is fit by optimizing MSE, while the error it prints in the report and what you get by calling `fg.get_params()` is actually MAE? 

I don't see an issue other than the inconsistency being aesthetically unpleasing. I assume the fitting part is correctly explained in the paper & docs. 

So yeah I'm okay with just changing the description to say MAE, or include a function for MSE as well cuz I imagine at some point **somebody** will want to use those error metrics for model selection, in which case providing MSE will just be convenient.

**Comment by TomDonoghue** [comment]
@rdgao : yep, what you described is right.

So, we could rename the function '`_calc_error`, and have both MAE & MSE available. A hidden parameter could set which one is used (as long as we are super clear this is a reporting error and not a model fit error) then if anyone does want to switch to be seeing MSE, they could. 

In this case, I would say we still default to reporting MAE. Partly, this is just because we are already doing so, and so I sorta want to keep reporting the same error (so an update doesnt change reported errors) unless we have a good reason to say we think we should be reporting MSE.

**Comment by TomDonoghue** [comment]
Okay, this update is now pushed to #152, the implementation of which can be reviewed there. I went with defaulting to MAE, but generalizing the error function, and adding a hidden parameter, to have other error metrics available. 

Unless anyone has any more debate points or points of interest, I'm to close this - but feel free to re-open for any more discussion, or check the fix in #152 .
